coderd/notifications/enqueuer.go

```diff
@@ -52,9 +52,15 @@ func NewStoreEnqueuer(cfg codersdk.NotificationsConfig, store Store, helpers tem
 	}, nil
 }
 
+// Enqueue queues a notification message for later delivery, assumes no structured input data.
+func (s *StoreEnqueuer) Enqueue(ctx context.Context, userID, templateID uuid.UUID, labels map[string]string, createdBy string, targets ...uuid.UUID) (*uuid.UUID, error) {
+	// "nil" data will be omitted while building the JSON payload.
```

Comment:
Can you please clarify this comment?
IMHO we should never omit keys from the payload, since webhook receivers may be expecting them to always be present.

Reply:
I agree, but wouldn't that be a problem with backward compatibility of the payload format? I _really_ don't want to implement magic to support both of them.

Reply:
You don't need to do anything for BC-compat; you just bump the version number of the payload by a minor version and you're done. We're not receiving this payload, only sending it, so we never have to unless marshal it.